File: backend/app/models/point_category.py
"""积分商品分类模型"""
from datetime import datetime

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, String, Text

from .base import Base
from .user import gen_uuid


class PointCategory(Base):
    """积分商品分类表 - 支持多级层级"""

    __tablename__ = "point_categories"

    id = Column(String(36), primary_key=True, default=gen_uuid)
    name = Column(String(50), nullable=False, comment="分类名称")
    description = Column(Text, nullable=True, comment="分类描述")
    parent_id = Column(String(36), ForeignKey("point_categories.id"),
                       nullable=True, index=True, comment="父分类ID")
    icon_url = Column(String(500), nullable=True, comment="分类图标URL")
    banner_url = Column(String(500), nullable=True, comment="分类横幅URL")
    level = Column(Integer, nullable=False, comment="层级(1/2/3)")
    sort_order = Column(Integer, default=0, nullable=False, comment="排序权重")
    is_active = Column(Boolean, default=True, nullable=False, comment="是否启用")

    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow,
                        onupdate=datetime.utcnow, nullable=False)

File: admin-web/src/api/pointCategory.ts
/**
 * 积分商品分类管理 API
 */
import { request } from '@/utils/request'

const PREFIX = '/admin/point-categories'

// ==================== Types ====================

export interface PointCategory {
  id: string
  name: string
  description: string | null
  parent_id: string | null
  icon_url: string | null
  banner_url: string | null
  level: number
  sort_order: number
  is_active: boolean
  created_at: string
  updated_at: string
  children?: PointCategory[] // Only in tree endpoint
}

export interface PointCategoryCreate {
  name: string
  level: number
  description?: string | null
  parent_id?: string | null
  icon_url?: string | null
  banner_url?: string | null
  sort_order?: number
  is_active?: boolean
}

export interface PointCategoryUpdate {
  name?: string
  description?: string | null
  icon_url?: string | null
  banner_url?: string | null
  sort_order?: number
  is_active?: boolean
}

// ==================== API Functions ====================

/** 获取分类列表（平铺） */
export function listPointCategories(params?: { active_only?: boolean }) {
  const { active_only = false } = params ?? {}
  const q = new URLSearchParams()
  q.set('active_only', String(active_only))
  return request<PointCategory[]>({
    url: `${PREFIX}?${q.toString()}`,
    method: 'GET',
  })
}

/** 获取分类树（层级结构） */
export function getCategoryTree(params?: { active_only?: boolean }) {
  const { active_only = true } = params ?? {}
  const q = new URLSearchParams()
  q.set('active_only', String(active_only))
  return request<PointCategory[]>({
    url: `${PREFIX}/tree?${q.toString()}`,
    method: 'GET',
  })
}

/** 获取单个分类详情 */
export function getPointCategory(categoryId: string) {
  return request<PointCategory>({
    url: `${PREFIX}/${categoryId}`,
    method: 'GET',
  })
}

/** 创建分类 */
export function createPointCategory(data: PointCategoryCreate) {
  return request<{ id: string }>({
    url: PREFIX,
    method: 'POST',
    data,
  })
}

/** 更新分类 */
export function updatePointCategory(categoryId: string, data: PointCategoryUpdate) {
  return request({
    url: `${PREFIX}/${categoryId}`,
    method: 'PUT',
    data,
  })
}

/** 删除分类 */
export function deletePointCategory(categoryId: string) {
  return request({
    url: `${PREFIX}/${categoryId}`,
    method: 'DELETE',
  })
}
